src/Framework/Framework/Compilation/Binding/TypeRegistry.cs

```diff
@@ -56,7 +62,7 @@ public TypeRegistry AddSymbols(IEnumerable<Func<string, Expression?>> symbols)
         [return: NotNullIfNotNull("type")]
         public static Expression? CreateStatic(Type? type)
         {
-            return type == null ? null : new StaticClassIdentifierExpression(type);
+            return type is { IsPublic: true } or { IsNestedPublic: true } ? new StaticClassIdentifierExpression(type) : null;
```

Comment:
IsNestedPublic is annoying, because it will AFAIK return true for `C2` in the following case:

```
internal class C1 {
   public class C2 { }
}
```

I think it would be worth it to also check the chain of declaring types.